pkg/sql/compile/compile.go

```diff
@@ -1266,7 +1266,7 @@ func (c *Compile) generateNodes(n *plan.Node) (engine.Nodes, error) {
 	if len(ranges) == 0 {
 		return nodes, nil
 	}
-	step := len(ranges) / len(c.cnList)
+	step := (len(ranges) / len(c.cnList)) + 1
 	if step <= 0 {
```

Comment:
Now step cannot be <= 0.    What is a step?

Reply:
ranges represents how many blocks there are in total, cnList represents how many cn nodes exist in total, and then step represents how many blocks are processed by each cn node.

Reply:
yes, now step cannot be <=0

Reply:
This code is wrong.

if ranges is 20, len(cnList) is also 20, you break the steps to 2, 2, 2, ..., 0, 0, 0 ...?

Should this be (len(ranges) + len(c.cnList) - 1) / len(c.cnList)?

Reply:
I assume the following 0, 0, ... will not crash our code.

Reply:
yes